src/Domain/CoordinateMaps/Frustum.cpp

```diff
@@ -137,10 +143,25 @@ std::array<tt::remove_cvref_wrap_t<T>, 3> Frustum::operator()(
   const ReturnType& xi = source_coords[0];
   const ReturnType& eta = source_coords[1];
   const ReturnType& zeta = source_coords[2];
-  const ReturnType cap_zeta =
-      with_projective_map_
-          ? (w_minus_ + w_plus_ * zeta) / (w_plus_ + w_minus_ * zeta)
-          : zeta;
+  ReturnType cap_zeta;
+  if (zeta_distribution_ == Distribution::Projective) {
+    cap_zeta = (w_minus_ + w_plus_ * zeta) / (w_plus_ + w_minus_ * zeta);
```

Comment:
Would it make sense to move this into the `Interval` map, so here in the frustum you don't have to switch between distributions at all? Then you can just do `cap_zeta = zeta_mapping(-1, 1, -1, 1, zeta_distribution, singularity_pos)`. So the `Interval` map handles the redistribution on the logical axis and the frustum just has to worry about the deformation of the logical cube.

Reply:
The projective map is not currently written in such a way as to map from [A, B] to [a, b], so it can't be easily ported over to Interval right now. It only maps [-1,1] to [-1. 1] right now, but I can make this change and address this comment in a later PR (I do agree with the suggested change though)

Reply:
ok